src/check/problem.zig

```diff
@@ -1340,12 +1342,13 @@ pub const ReportBuilder = struct {
 /// Entry points are `appendProblem` and `deepCopyVar`
 pub const Store = struct {
     const Self = @This();
+    const ALIGNMENT = 16;
 
-    problems: Problem.SafeMultiList,
+    problems: std.ArrayListAlignedUnmanaged(Problem, ALIGNMENT) = .{},
```

Comment:
The SafeMultiList was causing compilation issues due to `Problem` not being a POD type, the cause was that the compiler was trying to generate our new serialise methods but that doesn't work for non-POD types. I'm not sure why that wasn't breaking compilation before -- anyway I replaced it with an ArrayList here. @jaredramirez  can you please consider if this is ok.

Reply:
I don't think `MultiList` makes sense with tagged unions (if we want that, we should separate out the tag and the union into a `struct` with a `tag` field and a `union` field I think), so I think what we actually want here is a `SafeList`. (I'm also not sure why we'd need a special higher alignment requirement for this.)

Reply:
Oh, I just asked @jaredramirez about this

> I think originally in the store, I also had type snapshots! But then those needed to be moved into their own Store. So I don't think there's any strong reason that we need a problem store vs just an array list in check types or wherever!